### ISSUE TITLE
device_tracker.bluetooth_le_tracker not working with HASS.IO since introduction...?

### ISSUE THREAD

**Issue by Eina61** [home-assistant/hassio#235]
The ble tracker was introduced in about version 5.? but I haven't been able to get it working at all.

My logs look like this:

```
2017-10-27 22:59:01 ERROR (SyncWorker_11) [homeassistant.util.package] Unable to install package gattlib==0.20150805: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-2hdf7tbx/gattlib/
2017-10-27 22:59:01 ERROR (MainThread) [homeassistant.setup] Not initializing device_tracker.bluetooth_le_tracker because could not install dependency gattlib==0.20150805
2017-10-27 22:59:01 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform device_tracker.bluetooth_le_tracker: Could not install all requirements.
```

I have tried with a fresh hass.io install about 2 weeks ago but I still get the same errors in the logs.

I have Bluetooth BCM43xx (version 0.7) installed and it shows the following in the logs:

```
starting version 3.2.1
bcm43xx_init
Flash firmware /lib/firmware/brcm/BCM43430A1.hcd
Set Controller UART speed to 115200 bit/s
Device setup complete
```

I have tried it with & without this add-on. I am running on a Raspberry Pi 3.
Others are also reporting the same problem here:
https://community.home-assistant.io/t/hassio-list-bluetooth-le-devices/22798/17

**Comment by pvizeli** [comment]
You need add https://github.com/home-assistant/hassio-build/blob/master/homeassistant/generic/Dockerfile

But there no one was able to make it working. But if you have hardware for test it, feel free to make it working and make a PR 👍

**Comment by sharukins** [comment]
is pip install gattlib even working? This might be an upstream issue.
the compilation of gattlib throws an error when I try to install it on my laptop using python 3.6 and gcc 7.2

**Comment by sharukins** [comment]
Do we miss any dependencies for gattlib?

**Comment by pvizeli** [comment]
No but that person they have test it, was the meaning, that he don't work. Maybe while Alpine not run on glibc and need musl.

**Comment by pvizeli** [comment]
https://github.com/home-assistant/home-assistant/issues/9048

**Comment by sharukins** [comment]
It didn't work for him because he wasn't able to install gatttool which we have fixed now. Mi Flora now works, but the Bluetooth LE Tracker component needs gattlib which the guy succesfully installed.

**Comment by pvizeli** [comment]
If you want you can give a try

**Comment by sharukins** [comment]
Yeah, I'll give it a try, I'm just not sure how to test if it's working in the end

**Comment by Eina61** [comment]
Just an update after updating to 0.57.2 - still got the same problem. Log files says:

```
2017-11-07 19:48:12 ERROR (SyncWorker_5) [homeassistant.util.package] Unable to install package gattlib==0.20150805: Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-build-jcs9g68n/gattlib/
2017-11-07 19:48:12 ERROR (MainThread) [homeassistant.setup] Not initializing device_tracker.bluetooth_le_tracker because could not install dependency gattlib==0.20150805
2017-11-07 19:48:12 ERROR (MainThread) [homeassistant.setup] Unable to prepare setup for platform device_tracker.bluetooth_le_tracker: Could not install all requirements.
```

**Comment by sharukins** [comment]
I haven't had time to implement this

**Comment by jounivarjonen** [comment]
any news on this?

**Comment by sharukins** [comment]
sorry, but I need to finish my PHD thesis, therefore I lack time. Maybe someone else can jump in on this.

**Comment by JuanGrados** [comment]
Hi everyone! 

Are there any news in this issue? I have just updated today my HASSIO software to the latest version and I am also getting the same errors for BLE.

Thanks!! 
Cheers

**Comment by sharukins** [comment]
I fiddled around with this last weekend but no luck.
Maybe the BLE tracker should be completely rewritten to use bluepy as it was done for MiFlora. But that's way beyond my reach.

**Comment by pvizeli** [comment]
Some one need implement this to make it working: https://github.com/home-assistant/hassio/issues/235#issuecomment-340268886

**Comment by sharukins** [comment]
That's exactly what I tried, but it didn't work, gattlib throws errors on installation about missing libraries that I actually added.

**Comment by JuanGrados** [comment]
I see, that is a pity.
So, for my understanding, in principle, the fix implemented in Mi Flora platform could also be implemented to the tracker platform? 
Would not be better be implemented it in hassio it self to make the fix available to any other platform suffering the same problem? (probably my understanding of the different "levels" between hassio and the platforms is not the best) Just asking :)

**Comment by sharukins** [comment]
There is no fix for Mi Flora anymore, the component was completely rewritten so the answer to your first question is no.

**Comment by pvizeli** [comment]
It's would be fixable, you need only build the python module with correct library. It is not very difficult (Need only a bit knowledge about musl and glibc envs) but I have no time and to many task for do it self.

**Comment by sharukins** [comment]
Well then I'm obvously too stupid

**Comment by braddo99** [comment]
I'm too stupid as well, spent many hours trying to get BLE working with HASS (in Docker) going but massive errors always. Would so love this to work so I could use my beacons for something.

**Comment by disrupted** [comment]
since this seems to be a big issue I am thinking about taking a different route, just FYI @braddo99 https://github.com/mertenats/Open-Home-Automation/tree/master/ha_mqtt_binary_sensor_ble_scanner

**Comment by jaburges** [comment]
so just looking at this and it looks like firstly boost is needed but as per this thread:
https://bitbucket.org/OscarAcena/pygattlib/issues/41/ld-cannot-find-lboost_python-py34

lboost_python-py35 is there but the system looks for 34 - will report back

**Comment by jojoro1** [comment]
Any progress?

**Comment by nilssonac** [comment]
Any progress on this issue?

**Comment by allmoney** [comment]
Have same problem.

Hostname	hassio
System	Resin OS 2.3.0+rev1
Hass.io supervisor
Version	103.3

**Comment by noorbertt** [comment]
Any up?

**Comment by braddo99** [comment]
@ahmadalsaady When I see that error try:
sudo hciconfig hci0 reset    ... or whatever your device may be. I have never figured out why the device becomes unavailable to HA periodically but I did have two bluetooth devices, in built and dongle and when I took the dongle out, the issue presented much less frequently. Works pretty well for me now for BLE, still don't see bluetooth (nonLE) devices though.
